utils/schema/generate.js

```diff
@@ -0,0 +1,160 @@
+function createDefinitions(resources, lookups, additionalProperties = false) {
+  const typeMappings = {
+    'Edm.String': 'string',
+    'Edm.Boolean': 'boolean',
+    'Edm.Int16': 'integer',
+    'Edm.Int32': 'integer',
+    'Edm.Int64': 'integer',
+    'Edm.Decimal': 'number',
+    'Edm.Single': 'number',
+    'Edm.Double': 'number',
+    'Edm.DateTimeOffset': 'string',
+    'Edm.Date': 'string'
+  };
+
+  // Preprocess the lookups data to create a hashmap
+  const lookupsMap = {};
+  lookups.forEach(lookup => {
+    if (!lookupsMap[lookup.lookupName]) {
+      lookupsMap[lookup.lookupName] = new Set();
+    }
+    lookupsMap[lookup.lookupName].add(lookup.lookupValue);
+  });
+
+  function getPossibleValuesForNamespacedType(lookupName) {
+    const possibleValues = lookupsMap[lookupName];
+    return possibleValues ? [...possibleValues] : [];
+  }
+
+  const definitions = {};
+
+  for (const [resourceName, resourceFields] of Object.entries(resources)) {
+    const properties = {};
+
+    resourceFields.forEach(field => {
+      const { fieldName } = field;
+      let schema = {};
+      if (field.isExpansion || field.isCollection) {
+        const itemTypeSchema = {};
+
+        if (field.isExpansion) {
+          itemTypeSchema['$ref'] = `#/definitions/${field.typeName}`;
+        } else {
+          const mappedType = typeMappings[field.type] || 'object';
+          itemTypeSchema['type'] = mappedType;
+          if (field.maxLength) itemTypeSchema['maxLength'] = field.maxLength;
+          if (field.type === 'Edm.DateTimeOffset') itemTypeSchema['format'] = 'date-time';
+          if (field.type === 'Edm.Date') itemTypeSchema['format'] = 'date';
+
+          // probably needs to be another condition?
+          if (field.type.startsWith('org.reso.metadata')) {
+            const possibleValues = getPossibleValuesForNamespacedType(field.type);
+            if (possibleValues.length > 0) {
+              if (!additionalProperties) {
+                itemTypeSchema['enum'] = possibleValues;
+              }
+              itemTypeSchema['type'] = 'string';
+            }
+          }
+        }
+
+        if (field.isCollection) {
+          schema['type'] = 'array';
+          schema['items'] = itemTypeSchema;
+        } else {
+          schema = itemTypeSchema;
+        }
+
+        if (field.nullable) schema['nullable'] = field.nullable;
+      } else {
+        const fieldType = typeMappings[field.type] || 'object';
+        schema['type'] = fieldType;
+
+        if (field.nullable) schema['nullable'] = field.nullable;
+        if (field.maxLength) schema['maxLength'] = field.maxLength;
+        // do we need any special treatment for dates?
```

Comment:
Dates and timestamps should be validated against [ISO-8601 format](https://en.wikipedia.org/wiki/ISO_8601) (in both cases). 

It looks like this is already being done by using date and date-time https://json-schema.org/understanding-json-schema/reference/string.html#dates-and-times